sasdata/data_util/manipulations.py

```diff
@@ -282,8 +283,14 @@ class Binning:
     def __init__(self, min_value, max_value, n_bins, base=None):
         """
         if base is None: Linear binning
+        Todo: should base be a boolean? True = log? or do we want to allow for
+              other types of binning in the future, in which case we should
+              define base="linear" or "log" and explicitly set
+              if base = None then base="log"
         """
-        self.min = min_value if min_value > 0 else 0.0001
+        # min and max values can be negative.
+        # todo: do we want to check for min < max?
```

Comment:
Annular averaging gives the exact same result if `q_min > q_max`. There is no way to do this for sector cuts and slabs through the GUI. A unit test to show this is true for all slicer models could be useful...

Reply:
I am a bit confused about what point is bein made here. For a true annular (phi average) it makes no sense to have q < 0 (max or min) but otherwise does. I don't think flip gets called unless there is a limited phi range in which case allowing q < makes sense.  However I have noted that all slicer interactors seem to allow max < min and have been trying to figure out if that is a feature that is brilliant or a bug that needs squashing.  Is it the last that you are concerned about here? or something else?

Reply:
I'm saying the check you are suggesting, to see if `q_max > q_min`, isn't needed. Mathematically, there is no difference in the resulting calculation regardless of which is larger. I calculated the annular averaging slicer twice, where the min and max q values are swapped for the second calculation, and the results are exactly the same. I don't know if this is true for other slicers, but that's where my unit test suggestion would come into play. Run the calculations for every slicer twice as I did for the annular averaging then compare the results.

Reply:
Ok ... I see the confusion. Yes I know it works both ways my question was really "should it"? I have rewritten my comment to accurately reflect the question.